src/filter_tree/filter_tree.c

```diff
@@ -418,6 +418,155 @@ void FilterTree_DeMorgan(FT_FilterNode **root) {
 	_FilterTree_DeMorgan(root, 0);
 }
 
+// This method returns if an arithmetic expression is valid to be use in filter tree compaction.
+static inline bool _FT_FilterNode_constnat_expression(AR_ExpNode *exp) {
+	// TODO: Handle injected parames.
+	return exp->type == AR_EXP_OPERAND && exp->operand.type == AR_EXP_CONSTANT;
+}
+
+// Compacts an expression node. Return if this node can be used in compression.
+static inline bool _FilterTree_Compact_Exp(FT_FilterNode *node) {
+	return _FT_FilterNode_constnat_expression(node->exp.exp);
```

Comment:
There's no compaction applied here to the expression.

Reply:
fixed